clientd3d/statgame.c

```diff
@@ -197,16 +206,78 @@ void GameMouseButtonDown(HWND hwnd, BOOL fDoubleClick, int x, int y, UINT keyFla
    UserDidSomething();   /* User is alive! */
 
    SetFocus(hMain);
+
+   if (text_area_resize_zone) {
+       // Start text area resize.
+       text_area_resize_start_x = x;
+       text_area_resize_start_y = y;
+       text_area_resize_inprogress = true;
+   }
+
 }
 /****************************************************************************/
 void GameLButtonUp(HWND hwnd, int x, int y, UINT keyFlags)
 {
-   UserEndDrag();
+    if (text_area_resize_inprogress) {
+        // End text area resize.
+        text_area_resize_zone = false;
+        text_area_resize_inprogress = false;
+    }
+
+    UserEndDrag();
 }
 /****************************************************************************/
 void GameMouseMove(HWND hwnd, int x, int y, UINT keyFlags)
 {
-   //GameWindowSetCursor();
+    RECT rcToolbar;
+    ToolbarGetUnionRect(&rcToolbar);
+
+    /* Resize the text area is possible from the left hand edge of the client window,
+    in-between the viewport and textarea. The vertical resize cursor will be displayed
+    at this time. */
+    int offset = 5;
+    if (text_area_resize_inprogress ||
+        (y >= main_viewport_height + rcToolbar.bottom + EDGETREAT_HEIGHT- offset
+        && y <= main_viewport_height + EDGETREAT_HEIGHT + rcToolbar.bottom + offset
+        && x <= EDGETREAT_HEIGHT + offset))
+    {
+        MainSetCursor(hwnd, hwnd, HTTOP, 0);
+        text_area_resize_zone = true;
+    }
+    else
+    {
+        text_area_resize_zone = false;
+        text_area_resize_inprogress = false;
+    }
+
+    if (text_area_resize_inprogress)
+    {
+        RECT rect;
+        GetClientRect(hwnd, &rect);
+        int width = rect.right - rect.left;
+        int height = rect.bottom - rect.top;
+
+        // Calculate new text area size.
+        int delta_x = text_area_resize_start_x - x;
+        int delta_y = text_area_resize_start_y - y;
+        int new_text_area_height = (((float)config.text_area_size / 100.0) * height) + delta_y;
```

Comment:
This makes the mouse offset from the start of the drag into a velocity, rather than an absolute value.  As a result, if you move the mouse up a little and then stop, the graphics area continues to get smaller even though you're not doing anything.  I think this would work better if you used the current delta_x and delta_y (renamed) to calculate a new absolute size, rather than adding to the current size.  You might have to introduce variables to store the original sizes from the start of the drag to get that to work.